### ISSUE TITLE
Code completion broken in WebStorm

### ISSUE THREAD

**Issue by frysztak** [chakra-ui/chakra-ui#2265]
# Bug report

## Describe the bug

Code completion is broken in WebStorm.

## To reproduce

1. Create a minimal component:
```ts
import { Link as ChakraLink } from '@chakra-ui/core';
import React from 'react';

export interface LinkProps {
  isActive: boolean;
  to: string;
  children: string;
}

export function Link(props: LinkProps) {
  const { isActive, to, children } = props;

  return <ChakraLink>{children}</ChakraLink>;
}
```
2. Position cursor after `ChakraLink` but before `>`
3. Hit ctrl-space to trigger autocomplete
4. Result:
![image](https://user-images.githubusercontent.com/217806/96373613-fe456400-116d-11eb-8955-9713de79680f.png)

## Minimal reproduction

Minimal repro: https://codesandbox.io/s/peaceful-euler-s3p8n?file=/src/Link.tsx

## Expected behavior

IDE offers completion, like VSCode does:
![image](https://user-images.githubusercontent.com/217806/96373763-e7ebd800-116e-11eb-9363-6de7194e2f6d.png)

## System information

- OS: Win 10
- Browser (if applies): Firefox
- Version of @chakra-ui/core: 1.0.0-rc.5
- Version of Node.js: 12.18.1
- WebStorm: 2020.2.3
- TypeScript: 3.9.7

**Comment by ljosberinn** [comment]
Talked about this on discord with someone else literally as you filed the issue - fairly confident this is a WebStorm issue. Libraries cannot do anything special to enable/disable it.

**Comment by ljosberinn** [comment]
Try to invalidate Webstorms cache - the user on discord realized they previously had v0.8 installed and it works after that again. 

If this applies to you too, I'd recommend deleting node_modules and lockfile and to reinstall just to be safe.

**Comment by frysztak** [comment]
`yarn install --frozen-lockfile` and WebStorm restart seem to have done the trick. Thanks for a quick reply!

**Comment by Lulololu** [comment]
Hola,

The exact same thing happens to me with Webstorm. So far, I've tried : 

- DeletingNode Modules and package-lock.json - Invalidate Cache and Restart ;
- Installing Webstorm EAP 2020.3.

Neither solution worked for me.

So, I've opened an Issue on Webstorm Youtrack :

https://youtrack.jetbrains.com/issue/WEB-48085

Thank You,

**Comment by with-heart** [comment]
Thanks for opening that over there @Lulololu. Unfortunately I don't think anyone on the team uses WebStorm, so it's difficult for us to investigate. Some teammates at work use PyCharm so I'll see if they can look into it for me.

**Comment by Lulololu** [comment]
Thank You, with-heart.

At least, an Issue has been opened on Webstorm Youtrack and I hope we'll get some answer soon. 

Webstorm is really awesome, and Chakra-UI also, so....

**Comment by jongbelegen** [comment]
Any update on this? never experiences this problem with other UI component sets and this is keeping me from using your product

**Comment by TimKolberger** [comment]
I don't experience any issues with PhpStorm 2020.2 on macOS Big Sur.

![image](https://user-images.githubusercontent.com/16899513/101539863-74f92380-399f-11eb-8476-f492c9b19987.png)

**Comment by jongbelegen** [comment]
While `<Box />` seems to work for me too. `<Image />` and `<Input />` for example are broken on my machine.
but it was only available after some time and several invalidate cache restarts

**Comment by TimKolberger** [comment]
@jongbelegen This is correct! 

I can confirm missing autocomplete on `Image`, `Input`, `DrawerContent`, `ModalContent`, `Accordion`, etc.

@segunadebayo @with-heart I am looking for a pattern why which components work and others don't and I can only think of a too complex `as`-prop type interference.

I think we should look into this again.

**Comment by JohannesKlauss** [comment]
`Divider` component also has no autocomplete for me.

**Comment by dohomi** [comment]
I am following https://youtrack.jetbrains.com/issue/WEB-48085 because on my first hands on of chakra-ui I get no auto-import inside of WebStorm. I use v1.1.6 did I miss anything because this issue is closed even though for me the code-completion (auto-import) is not working correctly.

**Comment by with-heart** [comment]
Hi @dohomi, it's closed on our end because there's nothing for us to do for it. It's a problem with the JetBrains system.

**Comment by RIP21** [comment]
@frysztak @dohomi @JohannesKlauss and others.
In the latest release for WebStorm and other JetBrains IDEs, it's finally fixed! 2021.1 and onwards :)

**Comment by YuxLiu** [comment]
I had this issue using yarn2 in new JetBrains IDE , npm is fine.

> @frysztak @dohomi @JohannesKlauss and others.
> In the latest release for WebStorm and other JetBrains IDEs, it's finally fixed! 2021.1 and onwards :)
